### DISCUSSION TITLE
Configurable default wait strategy for DockerComposeEnvironment

### DISCUSSION THREAD

**Discussion by treydavis** [testcontainers/testcontainers-node#444] [Ideas]
I'm running into a small inconvenience related to using distroless containers with `DockerComposeEnvironment`. Distroless images don't work with the default port wait strategy. This means I have to use `withWaitStrategy` to override the strategy for my distroless container. However, I would like to keep specific container-related info out of my tests. For example, I'd like to avoid this:
``` js
const environment = await new DockerComposeEnvironment(
    composeFilePath,
    composeFile
).withWaitStrategy(
    'some-specific-container',
    Wait.forLogMessage(/.*some log output specific to the container.*/)
).up();
```
I'd rather create a `healthcheck` for `some-specific-container` in my Docker compose file so that all the container-specific configs are defined in one place, inside the container definition. 

This approach has led me to wish for a configurable default wait strategy:
``` js
const environment = await new DockerComposeEnvironment(
    composeFilePath,
    composeFile
).withDefaultWaitStrategy(Wait.forHealthCheck());
```

Or, perhaps, an alternative signature to `withWaitStrategy` (or a separate method name) that accepts a function mapping container to wait strategy:
``` typescript
const environment = await new DockerComposeEnvironment(
    composeFilePath,
    composeFile
).withWaitStrategy((container: Dockerode.Container) => {
    if (container.Labels... === "something") return Wait.forHttp(...);
    if (container.Labels... === "other") return Wait.forSomething();
    return Wait.forHealthCheck();
})
```

**Comment by cristianrgreco** [comment]
Hi @treydavis,

This is the default behaviour. If your containers define their own healthchecks, you simply need to tell Testcontainers to use them:

```js
const container = await new GenericContainer("my-container-with-health-check)
  .withWaitStrategy(Wait.forHealthCheck())
```

There is an issue related to this, which suggests Testcontainers should automatically switch to use `Wait.forHealthCheck()` if a container has a health check defined: https://github.com/testcontainers/testcontainers-node/issues/687
